### ISSUE TITLE
Assert that async `expect`s are awaited or returned

### ISSUE THREAD

**Issue by SimenB** [jest-community/eslint-plugin-jest#54]
`valid-expect` should check that if there's a `rejects` or `resolves` on an `expect`, that it's either `await`ed or `return`ed.

Example where we should warn:

```js
test('some test', () => {
  expect(Promise.resolve('hello')).resolves.toBe('hello');
});
```

**Comment by macklinu** [comment]
I'm going to give this one a try today.

**Comment by macklinu** [comment]
Some questions about `await` syntax - is this valid?

```js
test('some test', async () => {
  expect(await Promise.resolve('hello')).resolves.toBe('hello');
});
``

**Comment by SimenB** [comment]
> is this valid?

Nope, correct there is no `resolves` as it's already awaited

**Comment by macklinu** [comment]
I closed #78 - this is open for someone to give another shot at resolving this issue. 👍

**Comment by callumlocke** [comment]
This would be a great feature, would have saved me a lot of mistakes today.

It would be nice if it could also be configurable, to enforce one or the other syntax:

```js
// return
test('some test', () =>
  expect(Promise.resolve('hello')).resolves.toBe('hello');
);

// async/await
test('some test', async () => {
  await expect(Promise.resolve('hello')).resolves.toBe('hello');
});
```

Personally I would want to configure it to allow only `await` (even in functions with just a single assertion) because it's more explicit. With implicit return it's more likely someone (me) might come along later and add braces in order to add a second assertion, and inadvertently break it.

**Comment by SimenB** [comment]
I agree it would be best if we could enforce `await` over `.then` or `return`ing the promise
